src/main/java/com/bergerkiller/bukkit/tc/signactions/SignActionAnnounce.java

```diff
@@ -37,16 +37,21 @@ public boolean match(SignActionEvent info) {
 
     @Override
     public void execute(SignActionEvent info) {
-        String message = getMessage(info);
-        if (info.isTrainSign() && info.isAction(SignActionType.GROUP_ENTER, SignActionType.REDSTONE_ON)) {
-            if (!info.hasRailedMember() || !info.isPowered()) return;
-            sendMessage(info, info.getGroup(), message);
-        } else if (info.isCartSign() && info.isAction(SignActionType.MEMBER_ENTER, SignActionType.REDSTONE_ON)) {
-            if (!info.hasRailedMember() || !info.isPowered()) return;
-            sendMessage(info, info.getMember(), message);
-        } else if (info.isRCSign() && info.isAction(SignActionType.REDSTONE_ON)) {
-            for (MinecartGroup group : info.getRCTrainGroups()) {
-                sendMessage(info, group, message);
+        // check if Sign is REDSTONE_ON
+        // setup AnnounceMessage and useTitle if so
+        if (info.isAction(SignActionType.REDSTONE_ON)) {
+
+            AnnounceMessage message = new AnnounceMessage(info);
+            SignActionAnnounce.useTitle = isTitleAnnounce(info);
+
+            if (info.isTrainSign() && info.isAction(SignActionType.GROUP_ENTER)) {
+                if (!info.hasRailedMember() || !info.isPowered()) return;
+                sendMessage(info.getGroup(), message);
+            } else if (info.isCartSign() && info.isAction(SignActionType.MEMBER_ENTER)) {
+                if (!info.hasRailedMember() || !info.isPowered()) return;
+                sendMessage(info.getMember(), message);
+            } else if (info.isRCSign()) {
+                info.getRCTrainGroups().forEach(group -> sendMessage(group, message));
```

Comment:
@bergerkiller Is this slower than a plain for-loop? Will it create a significant slowdown if this is done?

Reply:
Nah that's fine, Collection.forEach isn't slower than a for loop, that's only an issue with stream api but even here that wouldn't have been a problem